### ISSUE TITLE
UI change - pages aligned on top

### ISSUE THREAD

**Issue by amivaleo** [junrrein/pdfslicer#68]
Hi,

this is only a very aesthetic "issue": what about aligned the open pdf pages on top?
Right now they're vertically aligned in the middle of the window.

It's just a bit weird for me, since this is the only app I know that use such alignment for a list of items. 🤷‍♂️

**Comment by junrrein** [comment]
Thanks for the report.

I agree with you. The issue in this case is that `GtkFlowBox` doesn't support doing what you have in mind. Other applications work around this issue by doing some hackery that I don't know if it can be managed with Gtkmm.

**Comment by amivaleo** [comment]
Uhm... I can't help with that. :(
My skills in Glade are just some drag&drop.

I hope you will find an easy and simple solution without putting too much effort.
It's not such a mandatory change, it's more about being consistent with similar object lists.

**Comment by junrrein** [comment]
I think I once saw an example for doing this, in Python, that I think can be translated for our situation. It's now a matter of finding that again.

**Comment by amivaleo** [comment]
I still struggle to understand how can you dev-guys "write" the UI...

**Comment by junrrein** [comment]
> I still struggle to understand how can you dev-guys "write" the UI...

To be honest, I should be using Glade to build my UI. "Writing" the UI is the "traditional" method, and it's what you learn first when learning Gtk. It's not complicated, it's just another way to do it.

**Comment by amivaleo** [comment]
How did you learn that? Do you have some book to suggest or some set of video-lessons?

**Comment by junrrein** [comment]
If you are completely new to programming, I would suggest you to learn something like Python first. It will save you a lot of frustrations compared to, for example, C++.

I don't really know about resources for learning Python, but I've heard good things about [Diving Into Python 3](https://www.diveinto.org/python3/index.html). I know even less about video lectures, since that's not the kind of material I prefer, but I'm sure there must be some good ones around.

An alternative is to look for a book or course that's about using Python for a specific domain you are interested in. For example, "Image processing with Python" or "Web development using Python".

After you have the basics of your language covered, you can turn to a tutorial about using Gtk in that language. The official resource in this area is [The Python GTK+ 3 Tutorial](https://python-gtk-3-tutorial.readthedocs.io/en/latest/). If it's anything like the Gtkmm Tutorial (the one I followed for C++), it will be quite good.

I'll be happy to answer any further questions you have.

**Comment by junrrein** [comment]
@amivaleo Hey there.

On reading this again, I think you were asking me specifically how one can build an UI using source code. The "Getting Started" section of the Python Gtk+3 Tutorial has an example (link [here](https://python-gtk-3-tutorial.readthedocs.io/en/latest/introduction.html)) where a couple of very simple UIs are created using code. I hope those can give you a better picture.

**Comment by amivaleo** [comment]
> You know Slicer is programmed in C++ right?

Yes. I opened the main.cpp file and that's it. But I recognized c++ in there.

My main concern is about how to make the code "talk" to the system and files. In other words: APIs, I guess.

It's a bunch of stuff to learn. I'll try to, in the spare time. And yes, I will email you. Thanks! :)

/offtopic

**Comment by junrrein** [comment]
@amivaleo 

> It's a bunch of stuff to learn. I'll try to, in the spare time.

If you want to brush up on C++ from the ground up, I can recommend the book ["Programming -- Principles and Practice Using C++ (Second Edition)"](http://www.stroustrup.com/Programming/), except the chapters related to the Calculator example (6 and 7), and the chapters about GUI stuff (which you can replace with the Gtkmm tutorial).

By all means, don't feel obligated or pressured in any way to do this.

Have a nice day!
